src/pybind/mgr/dashboard/tools.py

```diff
@@ -766,6 +771,18 @@ def str_to_bool(val):
     return bool(strtobool(val))
 
 
+def json_str_to_dict(value):  # type: (AnyStr) -> Dict
+    try:
+        return json.loads(value)
+    except ValueError as error:
+        logger.error(str(error))
+        return {}
+
+
+def partial_dict(orig, keys):  # type: (Dict, List[str]) -> Dict
```

Comment:
Could you please add some documentation. When using doctest as in ``str_to_bool`` you will also provide examples to the developer and you don't need to add additional unittest code.

Reply:
Same as above.

Reply:
Done.